### DISCUSSION TITLE
Issues with the deletion

### DISCUSSION THREAD

**Discussion by joaogl** [argoproj/argo-cd#8852] [Q&A]
I’m having an issue where ArgoCD when deleting the resources is getting stuck because it tries to delete the child’s first and then the parents. This works well for some cases but I have cases where this doesn’t work for instance, certificates.. it deletes the certificate but because the certificate request still exists it recreates the certificate. And it just stays there deleting and recreating :/

Is there a a way to specify an order or just tell Argo to delete it all at once or to use "force delete" instead of the normal delete?

Thanks!

**Comment by alexmt** [comment]
I think the closes available option is non-cascading delete. However in this case Argo CD will just keep resources in the cluster. If you want to automate it (e.g. from pipeline) and don't want to provide cluster access you might use argocd cli: `argocd app delete-resource <APP_NAME> --force`

**Reply by joaogl** [reply]
Yeah but this would solve my issue, I would have to do this manually everytime... for that I can also do from the UI :/ I wanted something automated without having to propagate credentials to the CI/CD to execute that command. it seems like a super basic thing that Argo should know how to deal with.

Should I open a improvement request then?

**Reply by Paragon1970** [reply]
Experiencing the same / similar issue, in my case the apps are deployed with an applicationset, eventually they do get deleted but this can take a long time, cert-manager goes into a bit of a frenzy and the ArgoUI slows down while doing so.
To compound the problem the underlying tls secrets are not removed and persist in the namespace, this causes cert-manager to attempt to recreate and complain, along with high cardinality to `apiserver_request_duration_seconds_bucket` (Prometheus)

I suspect these tls secrets (certs) remain so as to ease with cert request throttling limits that cert issuers (LetsEncrypt) enforce.
